MeshPrimitive::createBox : Added uvs

danieldresser-ie opened the pull request:
Got tired of not having UVs on the default cube in Gaffer.

These UVs match the default cube in Maya.

danieldresser-ie commented:
While I'm at it, added normals to cube and plane.  This is more just for discussion - adding UVs to the cube feels completely necessary and non-controversial.  The normals I don't feel as strongly about, but IE users have been asking for it, and it was easy to do.

johnhaddon commented on src/IECoreScene/MeshPrimitive.cpp:
```diff
@@ -560,13 +560,37 @@ MeshPrimitivePtr MeshPrimitive::createBox( const Box3f &b )
 	p.push_back( V3f( b.min.x, b.min.y, b.max.z ) );	// 6
 	p.push_back( V3f( b.min.x, b.max.y, b.max.z ) );	// 7
 
-	verticesPerFaceVec.resize( sizeof( verticesPerFace ) / sizeof( int ) );
```
It looks like we no longer use `verticesPerFaceVec` or `vertexIdsVec`, but they're still declared up top? If you're interested in simplifying this further, it would seem that with C++11 we can avoid the intermediate vector and just do `new IntVectorData( { 4, 4, 4, 4, 4, 4 } )`.

danieldresser-ie commented:
Removed unused vars, updated one test case, added support for indexed normals to IECoreAlembic::MeshWriter.

Should be good to go if all the tests pass.

johnhaddon commented:
LGTM
